enrich.concept Long->Integer ClassCastException

```
MATCH (n:Tag {value:"college"})
WITH n
CALL ga.nlp.enrich.concept({enricher:"conceptnet5", tag:n, depth:2, admittedRelationships:["RelatedTo","UsedFor","PartOf"], limit:50})
YIELD result
RETURN true
```

```
Failed to invoke procedure `ga.nlp.enrich.concept`: Caused by: java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
```

--- a/src/main/java/com/graphaware/nlp/dsl/request/ConceptRequest.java
+++ b/src/main/java/com/graphaware/nlp/dsl/request/ConceptRequest.java
@@ -183,7 +183,7 @@ public class ConceptRequest extends AbstractProcedureRequest {
             request.setFilterByLanguage((Boolean) conceptRequest.get(FILTER_BY_LANGUAGE_KEY));
         }
         if (conceptRequest.containsKey(LIMIT_KEY)) {
-            request.setResultsLimit( (int) conceptRequest.get(LIMIT_KEY));
+            request.setResultsLimit( ((Long) conceptRequest.get(LIMIT_KEY)).intValue());
         }
         if (conceptRequest.containsKey(ENRICHER_KEY)) {
             request.setEnricherName(conceptRequest.get(ENRICHER_KEY).toString());

--- a/src/main/java/com/graphaware/nlp/dsl/request/ConceptRequest.java
+++ b/src/main/java/com/graphaware/nlp/dsl/request/ConceptRequest.java
@@ -34,6 +34,7 @@ public class ConceptRequest extends AbstractProcedureRequest {
     public final static List<String> DEFAULT_ADMITTED_POS = Arrays.asList();
     private static final int DEFAULT_RESULTS_LIMIT = 100;
     private static final String DEFAULT_ENRICHER = ConceptNet5Enricher.ENRICHER_NAME;
+    private static final Double DEFAULT_MIN_WEIGHT = 0.0d;
 
     private int depth = DEFAULT_DEPTH;
     private String language = DEFAULT_LANGUAGE;
@@ -44,6 +45,7 @@ public class ConceptRequest extends AbstractProcedureRequest {
     private List<String> admittedPos = DEFAULT_ADMITTED_POS;
     private int resultsLimit = DEFAULT_RESULTS_LIMIT;
     private String enricherName = DEFAULT_ENRICHER;
+    private Double minWeight = DEFAULT_MIN_WEIGHT;
     
     private Node annotatedNode;
     private Node tag;
@@ -64,7 +66,8 @@ public class ConceptRequest extends AbstractProcedureRequest {
                 LIMIT_KEY,
                 TEXT_PROCESSOR_KEY,
                 LANGUAGE_KEY,
-                ENRICHER_KEY
+                ENRICHER_KEY,
+                MIN_WEIGHT
         );
     }
 
@@ -156,6 +159,14 @@ public class ConceptRequest extends AbstractProcedureRequest {
         this.enricherName = enricherName;
     }
 
+    public Double getMinWeight() {
+        return minWeight;
+    }
+
+    public void setMinWeight(Double minWeight) {
+        this.minWeight = minWeight;
+    }
+
     public static ConceptRequest fromMap(Map<String, Object> conceptRequest) {
 
         ConceptRequest request = new ConceptRequest();
@@ -187,6 +198,10 @@ public class ConceptRequest extends AbstractProcedureRequest {
         }
         if (conceptRequest.containsKey(ENRICHER_KEY)) {
             request.setEnricherName(conceptRequest.get(ENRICHER_KEY).toString());
+        }
+
+        if (conceptRequest.containsKey(MIN_WEIGHT)) {
+            request.setMinWeight((Double) conceptRequest.get(MIN_WEIGHT));
         }
 
         request.setProcessor((String) conceptRequest.get(TEXT_PROCESSOR_KEY));

--- a/src/main/java/com/graphaware/nlp/dsl/request/RequestConstants.java
+++ b/src/main/java/com/graphaware/nlp/dsl/request/RequestConstants.java
@@ -26,4 +26,5 @@ public final class RequestConstants {
     public static final String ANNOTATORS = "annotators";
     public static final String EXCLUDED_NER = "excludedNER";
     public static final String EXCLUDED_POS = "excludedPOS";
+    public static final String MIN_WEIGHT = "minWeight";
 }

--- a/src/main/java/com/graphaware/nlp/enrich/conceptnet5/ConceptNet5Enricher.java
+++ b/src/main/java/com/graphaware/nlp/enrich/conceptnet5/ConceptNet5Enricher.java
@@ -17,16 +17,13 @@ package com.graphaware.nlp.enrich.conceptnet5;
 
 import com.graphaware.common.log.LoggerFactory;
 import com.graphaware.common.util.Pair;
-import com.graphaware.nlp.configuration.DynamicConfiguration;
 import com.graphaware.nlp.domain.Tag;
 import com.graphaware.nlp.dsl.request.ConceptRequest;
 import com.graphaware.nlp.enrich.AbstractEnricher;
 import com.graphaware.nlp.enrich.Enricher;
 import com.graphaware.nlp.persistence.PersistenceRegistry;
-import com.graphaware.nlp.persistence.constants.Labels;
 import com.graphaware.nlp.processor.TextProcessor;
 import com.graphaware.nlp.processor.TextProcessorsManager;
-import com.graphaware.nlp.util.ProcessorUtils;
 import org.neo4j.graphdb.*;
 import org.neo4j.logging.Log;
 
@@ -99,7 +96,7 @@ public class ConceptNet5Enricher extends AbstractEnricher implements Enricher {
             }
         }
         tags.stream().forEach((tag) -> {
-            conceptTags.addAll(getImporter().importHierarchy(tag, lang, filterByLang, depth, processor, admittedRelationships, admittedPos, request.getResultsLimit()));
+            conceptTags.addAll(getImporter().importHierarchy(tag, lang, filterByLang, depth, processor, admittedRelationships, admittedPos, request.getResultsLimit(), request.getMinWeight()));
             conceptTags.add(tag);
         });
 

--- a/src/main/java/com/graphaware/nlp/enrich/conceptnet5/ConceptNet5Importer.java
+++ b/src/main/java/com/graphaware/nlp/enrich/conceptnet5/ConceptNet5Importer.java
@@ -61,18 +61,7 @@ public class ConceptNet5Importer {
         this.depthSearch = builder.depthSearch;
     }
 
-//    public List<Tag> importHierarchy(Tag source, String lang, boolean filterLang, int depth, TextProcessor nlpProcessor) {
-//        return importHierarchy(source, lang, filterLang, depth, nlpProcessor, DEFAULT_ADMITTED_RELATIONSHIP);
-//    }
-//
-//    public List<Tag> importHierarchy(Tag source, String lang, boolean filterLang, TextProcessor nlpProcessor) {
-//        return importHierarchy(source, lang, filterLang, depthSearch, nlpProcessor, DEFAULT_ADMITTED_RELATIONSHIP);
-//    }
-//
-//    public List<Tag> importHierarchy(Tag source, String lang, boolean filterLang, int depth, TextProcessor nlpProcessor, String... admittedRelations) {
-//        return importHierarchy(source, lang, filterLang, depth, nlpProcessor, Arrays.asList(admittedRelations));
-//    }
-    public List<Tag> importHierarchy(Tag source, String lang, boolean filterLang, int depth, TextProcessor nlpProcessor, List<String> admittedRelations, List<String> admittedPOS, int limit) {
+    public List<Tag> importHierarchy(Tag source, String lang, boolean filterLang, int depth, TextProcessor nlpProcessor, List<String> admittedRelations, List<String> admittedPOS, int limit, double minWeight) {
         if (null == admittedRelations || admittedRelations.isEmpty()) {
             throw new RuntimeException("Admitted Relationships is empty");
         }
@@ -87,6 +76,7 @@ public class ConceptNet5Importer {
                 values = client.queryByStart(finalWord, rel, lang, limit);
                 values.getEdges().stream().forEach((concept) -> {
                     if (checkAdmittedRelations(concept, admittedRelations)
+                            && concept.getWeight() > minWeight
                             && (concept.getStart().equalsIgnoreCase(source.getLemma()) || concept.getEnd().equalsIgnoreCase(source.getLemma()))
                             && (!filterLang || (filterLang && concept.getEndLanguage().equalsIgnoreCase(lang) && concept.getStartLanguage().equalsIgnoreCase(lang)))) {
 
@@ -103,7 +93,7 @@ public class ConceptNet5Importer {
                                     || posList.isEmpty()
                                     || posList.stream().filter((pos) -> (admittedPOS.contains(pos))).count() > 0) {
                                 if (depth > 1) {
-                                    importHierarchy(annotateTag, lang, filterLang, depth - 1, nlpProcessor, admittedRelations, admittedPOS, limit);
+                                    importHierarchy(annotateTag, lang, filterLang, depth - 1, nlpProcessor, admittedRelations, admittedPOS, limit, minWeight);
                                 }
                                 source.addParent(concept.getRel(), annotateTag, concept.getWeight(), ConceptNet5Enricher.ENRICHER_NAME);
                                 res.add(annotateTag);

--- a/src/test/java/com/graphaware/nlp/dsl/EnrichConceptProcedureTest.java
+++ b/src/test/java/com/graphaware/nlp/dsl/EnrichConceptProcedureTest.java
@@ -55,4 +55,20 @@ public class EnrichConceptProcedureTest extends NLPIntegrationTest {
             assertTrue(result.hasNext());
         }));
     }
+
+    @Test
+    public void testEnrichmentTakeMinWeightIntoAccount() {
+        clearDb();
+        executeInTransaction("CALL ga.nlp.annotate({text:'John and Adam went to college.', id: '123', textProcessor:'com.graphaware.nlp.stub.StubTextProcessor'})", (result -> {
+            assertTrue(result.hasNext());
+        }));
+
+        executeInTransaction("MATCH (n:AnnotatedText) CALL ga.nlp.enrich.concept({enricher: 'conceptnet5', node: n, depth: 1, language:'en', minWeight:100.0}) YIELD result RETURN result", (result -> {
+            assertTrue(result.hasNext());
+        }));
+
+        executeInTransaction("MATCH (n:Tag {value:'college'})-[:IS_RELATED_TO]->(x) RETURN x", (result -> {
+            assertFalse(result.hasNext());
+        }));
+    }
 }

--- a/src/test/java/com/graphaware/nlp/dsl/unit/ConceptRequestUnitTest.java
+++ b/src/test/java/com/graphaware/nlp/dsl/unit/ConceptRequestUnitTest.java
@@ -37,4 +37,18 @@ public class ConceptRequestUnitTest {
         map.put("admittedRelationship", new ArrayList<>());
         ConceptRequest request = ConceptRequest.fromMap(map);
     }
+
+    @Test
+    public void testRequestWithLimitLongCanBeCaster() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("tag", new NodeProxy());
+        map.put("admittedRelationships", Arrays.asList("IsA","PartOf"));
+        map.put("admittedPos", Arrays.asList("NNS","NN"));
+        map.put("language", "en");
+        map.put("splitTag", true);
+        map.put("limit", 50L);
+
+        ConceptRequest request = ConceptRequest.fromMap(map);
+        assertEquals(50, request.getResultsLimit());
+    }
 }

